failed to build on arm (Samsung Chromebook, Arch Linux)

I am trying to install the Brave browser on a Samsung Chromebook ARM machinne running Arch Linux.

I am using node@v5.6.0 and node-gyp@3.2.1 as per Brave's documentation, and the installation fails with an error because of abp-filter-parser-cpp, below is the output of me trying to npm install this on an empty folder:

```
[fcz@alarm test]$ npm install abp-filter-parser-cpp
npm WARN deprecated lodash.padright@3.1.1: This package has been renamed. Use lodash.padend@^4.0.0.

> abp-filter-parser-cpp@1.1.7 preinstall /home/fcz/test/node_modules/.staging/abp-filter-parser-cpp-a411e4a9
> npm install bloom-filter-cpp && npm install hashset-cpp

npm WARN deprecated lodash.padright@3.1.1: This package has been renamed. Use lodash.padend@^4.0.0.
npm ERR! Linux 3.8.11-3-ARCH
npm ERR! argv "/home/fcz/.nvm/versions/node/v5.6.0/bin/node" "/home/fcz/.nvm/versions/node/v5.6.0/bin/npm" "install" "bloom-filter-cpp"
npm ERR! node v5.6.0
npm ERR! npm  v3.6.0

npm ERR! Cannot read property 'target' of null
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /home/fcz/test/node_modules/.staging/abp-filter-parser-cpp-a411e4a9/npm-debug.log
/home/fcz/test
`-- (empty)

npm WARN enoent ENOENT: no such file or directory, open '/home/fcz/test/package.json'
npm WARN test No description
npm WARN test No repository field.
npm WARN test No README data
npm WARN test No license field.
npm ERR! Linux 3.8.11-3-ARCH
npm ERR! argv "/home/fcz/.nvm/versions/node/v5.6.0/bin/node" "/home/fcz/.nvm/versions/node/v5.6.0/bin/npm" "install" "abp-filter-parser-cpp"
npm ERR! node v5.6.0
npm ERR! npm  v3.6.0
npm ERR! code ELIFECYCLE

npm ERR! abp-filter-parser-cpp@1.1.7 preinstall: `npm install bloom-filter-cpp && npm install hashset-cpp`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the abp-filter-parser-cpp@1.1.7 preinstall script 'npm install bloom-filter-cpp && npm install hashset-cpp'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the abp-filter-parser-cpp package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm install bloom-filter-cpp && npm install hashset-cpp
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs abp-filter-parser-cpp
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls abp-filter-parser-cpp
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/fcz/test/npm-debug.log
npm ERR! code 1
```

I cloned the repo to try to debug it more, after cloning and npm install this is the error, better detailed:

```
> abp-filter-parser-cpp@1.1.7 install /home/fcz/github/abp-filter-parser-cpp
> node-gyp rebuild

make: Entering directory '/home/fcz/github/abp-filter-parser-cpp/build'
  CXX(target) Release/obj.target/abp-filter-parser-cpp/addon.o
  CXX(target) Release/obj.target/abp-filter-parser-cpp/ABPFilterParserWrap.o
  CXX(target) Release/obj.target/abp-filter-parser-cpp/ABPFilterParser.o
../ABPFilterParser.cpp:152:72: error: narrowing conversion of '-128' from 'int' to 'char' inside { } [-Wnarrowing]
 char separatorBuffer[32] = { 0, 0, 0, 0, 16, -128, 0, -92, 0, 0, 0, 64 };
                                                                        ^
../ABPFilterParser.cpp:152:72: error: narrowing conversion of '-92' from 'int' to 'char' inside { } [-Wnarrowing]
abp-filter-parser-cpp.target.mk:100: recipe for target 'Release/obj.target/abp-filter-parser-cpp/ABPFilterParser.o' failed
make: *** [Release/obj.target/abp-filter-parser-cpp/ABPFilterParser.o] Error 1
make: Leaving directory '/home/fcz/github/abp-filter-parser-cpp/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/fcz/.nvm/versions/node/v5.6.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:100:13)
gyp ERR! stack     at ChildProcess.emit (events.js:185:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 3.8.11-3-ARCH
gyp ERR! command "/home/fcz/.nvm/versions/node/v5.6.0/bin/node" "/home/fcz/.nvm/versions/node/v5.6.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/fcz/github/abp-filter-parser-cpp
gyp ERR! node -v v5.6.0
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok 

```

I am not a cpp programmer, but googling that error I got this stack overflow page that suggest a solution:

http://stackoverflow.com/a/31635045

I tested editing this line https://github.com/bbondy/abp-filter-parser-cpp/blob/master/ABPFilterParser.cpp#L152

to be:

```cpp
signed char separatorBuffer[32] = { 0, 0, 0, 0, 16, -128, 0, -92, 0, 0, 0, 64 };
```

And it fixed :)

--- a/ABPFilterParser.cpp
+++ b/ABPFilterParser.cpp
@@ -149,7 +149,7 @@ bool getFingerprint(char *buffer, const Filter &f) {
 }
 
 // Separator chars are one of: :?/=^;
-char separatorBuffer[32] = { 0, 0, 0, 0, 16, -128, 0, -92, 0, 0, 0, 64 };
+signed char separatorBuffer[32] = { 0, 0, 0, 0, 16, -128, 0, -92, 0, 0, 0, 64 };
 bool isSeparatorChar(char c) {
   return !!(separatorBuffer[(unsigned char)c / 8] & 1 << (unsigned char)c % 8);
 }

